lib/agent.js

```diff
@@ -32,7 +31,7 @@ class Agent extends EventEmitter {
     }
   }
 
-  get (origin) {
```

Comment:
Can we keep the `get`? It will be used for #587

Reply:
Not really... the api becomes weird. I can try to give it more thought.

Reply:
@mcollina I don't see how we can keep this... it just won't work with `RedirectAgent` and gives more api area that needs to be fulfilled...

Reply:
This was not a problem with the other implementation. Moreover this remove observability, as there is no way to know to access the Pool object.

I really think this should be retained.

Reply:
> This was not a problem with the other implementation

The other implementation is not correct IMHO.

Reply:
> Moreover this remove observability, as there is no way to know to access the Pool object.

I'm not sure I follow. What does this remove? Why would you need to access the Pool object?

Reply:
It makes sense to me to leave the dispatching of a request as an implementation detail of the agent.

Reply:
If we drop cross origin support it could make sense to keep the existing API. But that seems like a rather large thing to not support and kind of negates the point of having it inside of undici if people have to reimplement it anyway with cross origin support.

Reply:
I don't really understand what's the problem in exposing a get method.

Reply:
What would `RedirectAgent.get` return?

Reply:
`this[kAgent].get(origin)` would do.

Reply:
Note that exposing `get(origin)` enable somebody to access some stats about the `Pool` and monitor its state.

Reply:
Hm. I guess we could do that. One important note though is that `RediretPool.get(origin).request(opts)` won't follow any redirects. Which I think is kind of weird/confusing.

@dnlup also

Reply:
Adding @dnlup comment here so we have the whole conversation:

> Assuming we don't have to follow redirection using a specific origin, hence Pool (which I think it's unlikely, but I am just thinking out loud), I like this approach. I would expose the get method though, at this stage we need it and I think it causes no harm. I am open to revisiting it in the future (as I understand better what high-level and low-level APIs are needed by the user).

Reply:
I'm adding it back as @mcollina suggested. I'm not sure I think it's a good idea but it should not block this. I can open a separate PR where we can continue this discussion later.

Reply:
> Hm. I guess we could do that. One important note though is that RediretPool.get(origin).request(opts) won't follow any redirects. Which I think is kind of weird/confusing.

That would be totally ok for me as long as we document it.

Reply:
I understand your point, @ronag , and you're right. I wouldn't use a `Pool` instance for performing requests that need redirections. I agree that it wouldn't make sense. But `get(origin)` might get used for other low-level operations that do not require a request to be fired. As of now, I see more harm than good in removing it for that reason.

> Which I think is kind of weird/confusing.

On one side, it is. On the other one, if we think in therm of low-level API, it's just a small piece that does exactly one thing, and you might need it at that level. The end-user would use the high-level API and should not be concerned about it. I think we can document it well enough for someone who wants to implement a custom `Agent`. I am sure we can come up with a better approach in the future if needed.

Reply:
I know some of my colleagues that use `Agent.get` directly instead of the helper methods (they want to use dispatch). I could easily see that they would get confused and start using `RedirectAgent.get` thinking it does something it does not. Most people are also not very meticulous with reading the docs.

That's my main concern. It's kind of easy do things wrong which for me is a sign of a bad API. But anyways, let's discuss this in a separate PR. It's not that big of a problem.